### PR TITLE
sanitize hostname handling (fate #319639)

### PR DESCRIPTION
**Pull request by wfeldt**

### CODE REVIEW

File: file.c [hunk 1]
```diff
@@ -1895,13 +1895,6 @@ void file_write_install_inf(char *dir)
   if(config.net.realhostname) {
     file_write_str(f, key_hostname, config.net.realhostname);
   }
-  else {
-    char buf[256];
-    if(!gethostname(buf, sizeof buf)) {
-      log_debug("hostname = \"%s\"\n", buf);
-      if(*buf && strcmp(buf, "(none)")) file_write_str(f, key_hostname, buf);
-    }
-  }
```
Comment (mchf) [inline file.c]: What was purpose of this part. Why is it not needed anymore?

**Comment by mchf** [comment]
One question which is caused by lack of knowledge about linuxrc. Otherwise LGTM

File: file.c (hunk 1 above)
Comment (wfeldt) [inline file.c]: That was added as a temporary solution to pass on the wicked-set hostname and is no longer needed.

**Comment by mvidner** [comment]
LGTM. Thanks for the documentation!

**Pull request merged**